src/Domain/SizeOfElement.cpp

```diff
@@ -3,37 +3,39 @@
 
 #include "Domain/SizeOfElement.hpp"
 
-#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
-#include "DataStructures/Tensor/TypeAliases.hpp"
-#include "Domain/Mesh.hpp"  // IWYU pragma: keep
+#include "Domain/ElementMap.hpp"  // IWYU pragma: keep
 #include "Domain/Side.hpp"
-#include "ErrorHandling/Assert.hpp"
-#include "NumericalAlgorithms/LinearOperators/MeanValue.hpp"
-#include "NumericalAlgorithms/Spectral/Spectral.hpp"
 #include "Utilities/GenerateInstantiations.hpp"
 #include "Utilities/MakeArray.hpp"
+#include "Utilities/MakeWithValue.hpp"
 #include "Utilities/StdArrayHelpers.hpp"  // IWYU pragma: keep
 
 template <size_t VolumeDim>
 std::array<double, VolumeDim> size_of_element(
-    const Mesh<VolumeDim>& mesh,
-    const tnsr::I<DataVector, VolumeDim>& inertial_coords) noexcept {
-  ASSERT(mesh.quadrature() ==
-             make_array<VolumeDim>(Spectral::Quadrature::GaussLobatto),
-         "Implementation assumes that grid points extend to the faces of the\n"
-         "element, but not sure if this is true for quadrature: "
-             << mesh.quadrature());
+    const ElementMap<VolumeDim, Frame::Inertial>& element_map) noexcept {
   auto result = make_array<VolumeDim>(0.0);
-  // inertial-coord vector between lower face center and upper face center
-  auto center_to_center = make_array<VolumeDim>(0.0);
   for (size_t logical_dim = 0; logical_dim < VolumeDim; ++logical_dim) {
+    auto face_center =
+        [&logical_dim, &element_map ](const Side& side) noexcept {
+      const double dummy_for_size = 0.0;
+      auto logical_center =
+          make_with_value<tnsr::I<double, VolumeDim, Frame::Logical>>(
```

Comment:
given that you know the tensor type for this, do you need the `make_with_value`, or can you just call the tensor constructor?

Reply:
Indeed, this works. Thanks 🙂. I've put in a fixup for this and cleaned up some tiny things above and below as well.